figtree.go

```diff
@@ -506,6 +506,10 @@ func (m *Merger) makeMergeStruct(values ...reflect.Value) reflect.Value {
 					// unexported field, skip
 					continue
 				}
+				// For consistency with YAML data, determine a canonical field name based on the YAML tag.
+				// Do not rely on the Go struct field name unless there is no YAML tag.
+				field.Name = camelCase(yamlFieldName(field))
```

Comment:
I think this is going to be confusing.  If we merge this struct:
```
struct {
  MyXMLFile string `yaml:"my-xml-file"`
}
``` with this yaml document:
```
my-xml-file: foo.xml
```
then the resultant struct with be:
```
struct {
  MyXmlFile string
}
```
So the template would still have to be `{{.MyXmlFile}}` which now neither matches the input struct or the config file.

Reply:
But the same is true if you just consider the YAML document in isolation. Regardless of this proposed change, it would give you a `{{.MyXmlFile}}` template. What I am proposing here is using the yaml key as the source of truth for generating the template key. It makes it possible to have consistency between the struct and the YAML in your example, whereas without the change, it's impossible to merge that particular struct field with the YAML key.

I don't love this solution, but I like the status quo even less. I don't think we should force Go field names to follow strict rules, and I don't like how things are broken if we forget to follow those rules. I think it's not ideal that we're generating camelcased template key names that won't match YAML keys, but since that is the established convention we have to stick to, it makes sense to do it consistently both for untyped input (i.e. unmarshalled YAML) and structs.